azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+package azurerm
+
+// Code based on the terraform.provider plugin by Microsoft (R) AutoRest Code Generator.
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/hdinsight/mgmt/2015-03-01-preview/hdinsight"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmHDInsightClusters() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHDInsightClustersCreate,
+		Read:   resourceArmHDInsightClustersRead,
+		Update: resourceArmHDInsightClustersUpdate,
+		Delete: resourceArmHDInsightClustersDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_type": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"hadoop",
+					"hbase",
+					"storm",
+					"spark",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"cluster_version": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"component_version": {
+				Optional: true,
+				ForceNew: true,
+				Type:     schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"connectivity_endpoints": {
+				Computed: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"location": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"name": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"port": {
+							Computed: true,
+							Type:     schema.TypeInt,
+						},
+						"protocol": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+			"location": locationSchema(),
+			"name": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"os_type": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Linux",
+					"Windows",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"restauth_username": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"restauth_password": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"head_node": {
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"linux_os_profile": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Optional: true,
+										ForceNew: true,
```

Comment:
this won't be returned by the API - as such `linux_os_profile` will need to be a Set rather than a List with a hash function excluding this field; in addition this needs to be `Sensitive: true`